scanner/identifier.go

```diff
@@ -0,0 +1,93 @@
+package scanner
+
+import (
+	"fmt"
+
+	"github.com/grafana/river/token"
+)
+
+// IsValidIdentifier returns true if the given string is a valid river
+// identifier.
+func IsValidIdentifier(in string) bool {
+	s := New(nil, []byte(in), nil, 0)
+	_, tok, lit := s.Scan()
+	return tok == token.IDENT && lit == in
+}
+
+type SanitizeIdentifierOptions struct {
+	// Prefix is what will be prepended to the identifier if it does not start
+	// with a letter or underscore. This must be a valid river identifier.
+	Prefix string
+
+	// Replacement is what will be used to replace invalid characters. This
+	// must be a valid river identifier or empty.
+	Replacement string
+}
```

Comment:
As discussed offline, I think SanitizeIdentifier should be opinionated about how the sanitization works and accept no options, at least for now.